### PR TITLE
Better linting

### PR DESCRIPTION
**Pull request by jhabdas**
Adds more opinion to the way we lint and removes the Gitter badge as it's not adding enough value.

### CODE REVIEW

**Comment by jhabdas** [comment]
@jblock mind taking a look?

File: brunch-config.coffee [hunk 1]
```diff
@@ -32,5 +28,17 @@ exports.config =
...
         max_line_length: {level: 'ignore'}
+        cyclomatic_complexity: {level: 'warn'}
+        no_empty_param_list: {level: 'error'}
+        no_interpolation_in_single_quotes: {level: 'error'}
+        no_unnecessary_double_quotes: {level: 'warn'}
+        spacing_after_comma: {level: 'warn'}
+        ensure_super:
+          module: 'coffeelint-ensure-super'
+          check:
+            View: ['attach']
```
Comment (jblock) [inline brunch-config.coffee]: I think we should also have `Collection` and `Model`'s initialize here. The performance penalty should be minimal

File: app/views/base/view.coffee [hunk 2]
```diff
@@ -3,8 +3,10 @@ require('lib/view-helper') # Just load the view helpers, no return value
 module.exports = class View extends Chaplin.View
   autoRender: yes
 
-  # Precompiled templates function initializer.
+  # Precompiled templates function initializer
   getTemplateFunction: ->
+    # Register view-defined Handlebars partials
+    Handlebars.registerPartial(name, template) for name, template of @partials if @partials
```
Comment (jblock) [inline app/views/base/view.coffee]: Should we have a complementary line in `dispose` that calls:

```CoffeeScript
delete Handlebars.partials[partial] for partial in @partials if @partials
```

?

Comment (jhabdas) [inline app/views/base/view.coffee]: Done. I used the Handlebars API to unregister rather than calling `delete`.

File: brunch-config.coffee (hunk 1 above)
Comment (jhabdas) [inline brunch-config.coffee]: Done.

File: app/views/base/view.coffee (hunk 2 above)
Comment (jblock) [inline app/views/base/view.coffee]: Is the version of Handlebars loaded in BWP v2.0.0 or greater? from the looks of it it's not in anything below that: https://github.com/wycats/handlebars.js/issues/669

**Comment by jblock** [comment]
Other than the Handlebars question, LGTM

**Comment by jhabdas** [comment]
All set. Nice catch on the API. We're currently using Handlebars 1.x so used `delete` instead.

**Pull request merged**